src/test/java/com/facebook/presto/hadoop/TestLineReader.java

```diff
@@ -0,0 +1,173 @@
+package com.facebook.presto.hadoop;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.util.LineReader;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestLineReader
+{
+    @Test
+    public void testDefaultReaderZeroBytes()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4);
+        Text str = new Text();
+        try {
+            // There should not be any bytes read into str because the maxLineLength is 0
+            reader.readLine(str, 0, 30);
+            assertEquals(str.getLength(), 0);
+            assertEquals(str.getBytes(), "".getBytes());
+        }
+        catch (IOException e) {
```

Comment:
This catch is not needed. Just declare the checked exception for the test method. If it throws, the test will fail.